app/domain/authentication/authn_k8s/authenticator.rb

```diff
@@ -0,0 +1,300 @@
+module Authentication
+  module AuthnK8s
+    class AuthenticationError < RuntimeError; end
+    class CSRVerificationError < RuntimeError; end
+    class ClientCertVerificationError < RuntimeError; end
+    class ClientCertExpiredError < RuntimeError; end
+    class NotFoundError < RuntimeError; end
+    
+    class Authenticator
+      def initialize(env:)
+        @env = env
+      end
+
+      def inject_client_cert(params, request)
+        @params = params
+        @request = request
+        
+        verify_enabled
+        service_lookup
+        host_lookup
```

Comment:
Make this a verb phrase to be consistent with the naming conventions and the other methods in this list.  Ie, `lookup_host` rather than what sounds like a noun, `host_lookup`.  _however_, based on what the method actually does, i'd say `verify_host_exists` would be more accurate.